src/libraries/System.Security.Cryptography.Csp/src/System.Security.Cryptography.Csp.csproj

```diff
@@ -131,4 +131,7 @@
     <Reference Include="System.Threading" />
     <Reference Include="System.Threading.Thread" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Include="$(CommonPath)System\Obsoletions.cs" Link="Common\System\Obsoletions.cs" />
+  </ItemGroup>
```

Comment:
This `<Compile>` should only be needed within the `<ItemGroup Condition="'$(TargetsAnyOS)' != 'true'">` group above, as that looks like the only config where the affected types are added in.

I saw the same thing in https://github.com/dotnet/runtime/pull/52373/files#diff-0393b3204abdf640a38c160a004161f64402a8a739ea517bc2f4e7e35e3c76e4, but that one is not merged yet.

Reply:
This was done by analogy with
https://github.com/dotnet/runtime/blob/7ea26e68f9856df48fb6f48666fc643bbcbefbc6/src/libraries/System.Runtime.InteropServices/src/System.Runtime.InteropServices.csproj#L49

Reply:
Sorry; my comment was unclear. We do need to have this `<Compile>` added into the project file, but it shouldn't need to apply for all build configurations. We only need it to be applied within the build configuration that includes compiling the types that are referencing the `Obsoletions` members.  The `System.Runtime.InteropServices.csproj` file doesn't have the same multi-targeting scenario.

https://github.com/dotnet/runtime/pull/52373/files#diff-0393b3204abdf640a38c160a004161f64402a8a739ea517bc2f4e7e35e3c76e4 illustrates the correct change to make.

Reply:
Ah, got it :)